### PR TITLE
CAD and other stuffs

### LINKED ISSUE #29: Compress the TOC into one collapsed line
## Summary

I want to suggest to compress the table of contents into a one-liner.
I saw this on some github projects and I think this gives a big readme file some advantages:

- A tidier look
- If a string is searched the TOC can be excluded from search by collapsing it
- If the readme is used as a webpage (html conversion, e. g. when using github pages) and the user wants to print the page the TOC can be excluded from the printing output by collapsing it

## This is the current state

 * [Point 1](#point1)
    * [Point 1.1](#point11)
    * [Point 1.2](#point12)
 * [Point 2](#point2)
    * [Point 2.1](#point21)
    * [Point 2.2](#point22)

##  This is the suggestion
<details>
 <summary>Table of Contents</summary>

 * [Point 1](#point1)
    * [Point 1.1](#point11)
    * [Point 1.2](#point12)
 * [Point 2](#point2)
    * [Point 2.1](#point21)
    * [Point 2.2](#point22)
</details>

## This is the code
```markdown
<details>
 <summary>Table of Contents</summary>

 * [Point 1](#point1)
    * [Point 1.1](#point11)
    * [Point 1.2](#point12)
 * [Point 2](#point2)
    * [Point 2.1](#point21)
    * [Point 2.2](#point22)
</details>
```

### PR DESCRIPTION
**Pull request by Clockmaker**
Closes #29
Lots of commits, but they are mostly undo/reverts/various errors.

I used @pwab 's suggestion and added a collapsible ToC and "Back to Top" links

Other stuff:
* Plasticity section
* CAD software update
* CAD/CAM section update (theory/books/tutorials/techniques)

A note on ISBN style: 
* book title, etc... &emsp;&bull; <sub><sup>ISBN: 978-1439802472</sup></sub>

I personally like it, not much to say; if you dislike it, I can revert it back.
The html is this:
````html
&emsp;&bull; <sub><sup>ISBN: 978-1439802472</sup></sub>
````

To sum it up, the list needs a *style's guideline*.

### CODE REVIEW

**Review by pwab — APPROVED** [review]
Good job as always! 👍 
I don't like the &bull; (`&bull;`) in the ISBN markup but whatever.

**Comment by pwab** [comment]
**Just a little tip:**
You can add a `closes #xyz` into your PR ~title~ body which will automatically close the corresponding issue when the PR is merged. (Also works with `fix` and `resolve`)

~Example for the title:
CAD and other stuffs - Closes #29~

**Further reference:**
https://github.com/blog/1506-closing-issues-via-pull-requests

**Comment by Clockmaker** [comment]
thank you for the tip!

killing `&bull;` is a matter of seconds; refactoring ``<sub><sup>`` is the real annoying part.
So i guess it's somewhat fine, for now...

**Comment by pwab** [comment]
Argh I was telling nonsense - you have to include it into the body of your PR...

> All you have to do is include the special keyword syntax (eg. "fixes #5") in the **body of your Pull Request**.

So the title can stay as it was but only the body must include the signal word combined with the issue reference.

Sorry :sweat_smile:

**Comment by Clockmaker** [comment]
No problem
